crates/composer_api/src/lib.rs

```diff
@@ -1,3 +1,5 @@
+#![warn(clippy::all, clippy::clone_on_ref_ptr)]
```

Comment:
Yeah one needs to repeat this for every compilation unit root. https://github.com/rust-lang/rfcs/pull/3389 will change that fortunately.

Reply:
In CI we treat clippy warnings as errors (failing the CI job). That makes me ponder: shouldn't we set this to `deny` instead of `warn` so that locally we get the same semantics?

OTOH, I think it's quite easy to remember that I need to fix all warnings before pushing.

Reply:
> That makes me ponder: shouldn't we set this to deny instead of warn so that locally we get the same semantics?

I fear that may get a bit problematic during prototyping (where you regularly get clippy warnings like unused imports). I for one use clippy (called by my IDE) to see "does this still compile"? If we set `deny` here I would lose the distinction between "compiles with warnings" and "does not compile".

I guess a pre-commit hook with `cargo clippy --all-targets --all-features -- -D warnings` would be most appropriate here (assuming code is already more polished when about to commit).

Reply:
True! The usage of clippy during prototyping didn't appear to me.